src/error.rs

```diff
@@ -0,0 +1,26 @@
+use eyre::Report;
+use std::error::Error;
+
+use tlsn_notary::{NotaryConfigBuilderError, NotaryError};
+
+#[derive(Debug, thiserror::Error)]
```

Comment:
Ah I wanted to add an extra message for some error types, e.g. "Error occurred during notarization: {error message from internal lib}" for `NotaryError` — so that it's easier to differentiate errors in logs. `thiserror` makes it easy to keep these extra messages defined at a single place.

Also I foresee that we will have error types that we want to return to prover client, e.g. auth error, which will be nice to have its custom message and conversion logic from different internal errors encapsulated in an enum.